### DISCUSSION TITLE
Exceptions in @Scheduled methods

### DISCUSSION THREAD

**Discussion by ghost** [quarkusio/quarkus#34651] [Q&A] [answered]
What happens if an exception is thrown in a @Scheduled method? Is the task stopped and all future invocations stopped too? I couldn't find an explanation in a scenario like that. There seems to be a FailedExecution event that is thrown. Can that be handled? If so, how?

[a comment by an automated account is not shown]

**Reply by mkouba** [reply, marked as answer]
> What happens if an exception is thrown in a @scheduled method? Is the task stopped and all future invocations stopped too? 

Nope, it is not stopped.

We log a message with level `ERROR` and the text that starts with `Error occurred while executing task for trigger...`.
 
Furthermore, we fire the `FailedExecution` CDI event, both synchronously and asynchronously. You can observe these events and inspect the cause:

```java
import jakarta.enterprise.event.Observes;
import jakarta.enterprise.context.ApplicationScoped;

@ApplicationScoped
class MyBean {

   void onFailureSync(@Observes FailedExecution failure) {
      // do something with failure.getException()
   }
}

```
